NitroxClient/GameLogic/Spawning/WorldEntities/DefaultWorldEntitySpawner.cs

```diff
@@ -18,55 +23,138 @@ public IEnumerator SpawnAsync(WorldEntity entity, Optional<GameObject> parent, E
+        {
             gameObject.transform.position = entity.Transform.Position.ToUnity();
             gameObject.transform.rotation = entity.Transform.Rotation.ToUnity();
             gameObject.transform.localScale = entity.Transform.LocalScale.ToUnity();
 
             NitroxEntity.SetNewId(gameObject, entity.Id);
             CrafterLogic.NotifyCraftEnd(gameObject, techType);
 
-            if (parent.HasValue && !parent.Value.GetComponent<LargeWorldEntityCell>())
+            WaterPark parentWaterPark = parent.HasValue ? parent.Value.GetComponent<WaterPark>() : null;
+            if (!parentWaterPark)
             {
-                LargeWorldEntity.Register(gameObject); // This calls SetActive on the GameObject
+                if (parent.HasValue && !parent.Value.GetComponent<LargeWorldEntityCell>())
+                {
+                    LargeWorldEntity.Register(gameObject); // This calls SetActive on the GameObject
+                }
+                else if (gameObject.GetComponent<LargeWorldEntity>() && !gameObject.transform.parent && cellRoot.liveRoot)
+                {
+                    gameObject.transform.SetParent(cellRoot.liveRoot.transform, true);
+                    LargeWorldEntity.Register(gameObject);
+                }
+                else
+                {
+                    gameObject.SetActive(true);
+                }
             }
-            else if (gameObject.GetComponent<LargeWorldEntity>() && !gameObject.transform.parent && cellRoot.liveRoot)
+
+            if (parent.HasValue)
             {
-                gameObject.transform.SetParent(cellRoot.liveRoot.transform, true);
-                LargeWorldEntity.Register(gameObject);
+                if (parentWaterPark && gameObject.TryGetComponent(out Pickupable pickupable))
+                {
+                    pickupable.SetVisible(false);
+                    pickupable.Activate(false);
+                    parentWaterPark.AddItem(pickupable);
+                }
+                else
+                {
+                    gameObject.transform.SetParent(parent.Value.transform, true);
+                }
             }
-            else
+        }
+
+        public static bool TryGetCachedPrefab(out GameObject prefab, TechType techType = TechType.None, string classId = null)
+        {
+            if ((classId != null && prefabCacheByClassId.TryGetValue(classId, out prefab)) ||
+                (techType != TechType.None && prefabCacheByTechType.TryGetValue(techType, out prefab)))
             {
-                gameObject.SetActive(true);
+                return true;
             }
+            prefab = null;
+            return false;
+        }
 
-            if (parent.HasValue)
+        public static IEnumerator RequestPrefab(TechType techType, TaskResult<GameObject> result)
+        {
+            if (prefabCacheByTechType.TryGetValue(techType, out GameObject prefab))
```

Comment:
UWE.PrefabDatabase still requires you to execute that yield instruction which is what I want to avoid here. This costs a non-required time for batch spawning.
Pumping a coroutine isn't possible when it contains prefab loading instructions as the one used here.